### ISSUE TITLE
Error mounting SDCARD in micropython v1.24.0 ESP32_GENERIC

### ISSUE THREAD

**Issue by dpgeorge** [micropython/micropython#16214]
### Discussed in https://github.com/orgs/micropython/discussions/16125

<div type='discussions-op-text'>

<sup>Originally posted by **AledelaCV** November  1, 2024</sup>
I have upgraded to version 1.23.0 to 1.24.0 encountering the error when mounting SDCard. The sample code in the previous version works without problem, but in the new version of micropython it generates error: 
```
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
OSError: [Errno 5] EIO
```

Here's the code I'm using:
```
import machine, os, vfs

# Slot 2 uses pins sck=18, cs=5, miso=19, mosi=23
sd = machine.SDCard(slot=2)
vfs.mount(sd, '/sd') # mount

print(os.listdir('/sd'))    # list directory contents
```

This code works perfectly in version 1.23.0</div>

**Comment by dpgeorge** [comment]
Also possibly on stm32 (via @iabdalkader):

This commit broke something, the following code no longer works.
```
paste mode; Ctrl-C to cancel, Ctrl-D to finish
=== import pyb
=== sdcard = pyb.SDCard()
=== vfs = os.VfsFat(sdcard)
=== os.mount(vfs, "/sdcard")
=== 
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
OSError: [Errno 5] EIO
```

The same code works on 1.23.0 or without this commit

**Comment by dpgeorge** [comment]
Related #16171.

**Comment by dpgeorge** [comment]
The bug here does not manifest on stm32 by default (ie when you just let the SD card be mounted by the start-up code) because that uses `MP_BLOCKDEV_FLAG_NATIVE`.  The bug only manifests on stm32 when you manually access the SD card from Python.

**Comment by dpgeorge** [comment]
See #16222 and #16223 for fixes.

**Comment by dpgeorge** [comment]
Fixed by 5dc9eda1953668eb6861be01ca85f147dcf8d406
